volatility/framework/plugins/windows/envars.py

```diff
@@ -0,0 +1,155 @@
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [requirements.TranslationLayerRequirement(name = 'primary',
+                                                         description = 'Memory layer for the kernel',
+                                                         architectures = ["Intel32", "Intel64"]),
+                requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+                requirements.ListRequirement(name = 'pid',
+                                             description = 'Filter on specific process IDs',
+                                             element_type = int,
+                                             optional = True),
+                requirements.BooleanRequirement(name='SILENT',
+                                             description='Suppress common and non-persistent variables',
+                                             optional=True),
+                requirements.PluginRequirement(name = 'pslist', plugin = pslist.PsList, version = (1, 0, 0)),
+                requirements.PluginRequirement(name = 'hivelist', plugin = hivelist.HiveList, version = (1, 0, 0))
+                ]
+
+    def _get_silent_vars(self) -> List[str]:
+        """Enumerate persistent & common variables.
+        
+        This function collects the global (all users) and 
+        user-specific environment variables from the 
+        registry. Any variables in a process env block that
+        does not exist in the persistent list was explicitly
+        set with the SetEnvironmentVariable() API.
+        """
+
+        values = []
+
+
+        for hive in hivelist.HiveList.list_hives(context = self.context,
+                                 base_config_path = self.config_path,
+                                 layer_name = self.config['primary'],
+                                 symbol_table = self.config['nt_symbols'],
+                                 hive_offsets = None):
+            sys = False
+            ntuser = False
+
+            ## The global variables 
+            try:
+                key = hive.get_key('CurrentControlSet\\Control\\Session Manager\\Environment')
+                sys = True
+            except KeyError:
+                try:
+                    key = hive.get_key('ControlSet001\\Control\\Session Manager\\Environment')
+                    sys = True
+                except KeyError:
+                    pass
+            if sys:
+                try:
+                    for node in key.get_values():
+                        try:
+                            value_node_name = node.get_name()
+                            if value_node_name:
+                                values.append(value_node_name)
+                        except (exceptions.InvalidAddressException, RegistryFormatException) as excp:
+                            continue

```

Comment:
Again, when things happen we don't expect, it'd be good to give the user a way to find out about it (even if it's hidden in debug logs)  5:)